src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -565,6 +597,24 @@ def update_dashboard_filters(self, instance, validated_data):
             instance.filters = filters
             instance.save()
 
+    def update_permissions(self, instance, validated_data):
+        if "permissions" in validated_data and validated_data["permissions"] is not None:
+            permissions_data = validated_data["permissions"]
+            permissions = DashboardPermissions.objects.update_or_create(
+                dashboard=instance,
+                defaults={
+                    "is_editable_by_everyone": permissions_data["is_editable_by_everyone"],
+                },
+            )[0]
+            if "teams_with_edit_access" in permissions_data:
+                teams_data = permissions_data["teams_with_edit_access"]
+                if teams_data == [] or permissions_data["is_editable_by_everyone"] is True:
+                    permissions.teams_with_edit_access.clear()
+                else:
+                    permissions.teams_with_edit_access.set(Team.objects.filter(id__in=teams_data))
```

Comment:
At this point has `teams_data` been run through validation to ensure that only teams within the current organization are referenced?

Reply:
Yes, but added an org filter here as well